src/ILCompiler.Compiler/src/CppCodeGen/DependencyNodeIterator.cs

```diff
@@ -28,22 +28,13 @@ public DependencyNodeIterator(IEnumerable<DependencyNode> nodes, NodeFactory fac
                 if (node is EETypeNode)
                 {
                     var typeNode = node as EETypeNode;
-
-                    if (!_typeToNodeMap.ContainsKey(typeNode.Type))
-                        _typeToNodeMap[typeNode.Type] = typeNode;
-                    else if (typeNode is ConstructedEETypeNode)
-                        _typeToNodeMap[typeNode.Type] = typeNode;
-                }
-                else
-                if (node is CppMethodCodeNode)
-                {
-                    // TODO: Remove once the type depedencies are tracked properly
-                    // Ensure that there is going to be a forward type definition for the method containing type
-                    var methodNode = node as CppMethodCodeNode;
-
-                    var methodType = methodNode.Method.OwningType;
-                    if (!_typeToNodeMap.ContainsKey(methodType))
-                        _typeToNodeMap[methodType] = (EETypeNode)factory.NecessaryTypeSymbol(methodType);
+                    if (typeNode != null)
```

Comment:
This null check is not necessary. You have just checked that the node type is EETypeNode in the line above. It means that you will never get null here.